### PR TITLE
Feat html t2006

### PR DESCRIPTION
**Pull request by Abishek666**
#Description:
To use http APIs to fetch the city weather details through get and post APIs

#Implementation:
 - Added `fetchData.js` with API methods
 - Updated the prototypes to use methods from fetchData.js

### CODE REVIEW

File: src/baseClass.js [hunk 1]
```diff
@@ -41,4 +45,10 @@ export const changeHeaderValues = (cityName) => {
   city.updateTopMiddle()
   city.setTopDate()
   city.setTopTime()
+
+  setInterval(() => {
+    city.updateTopMiddle()
+    city.setTopDate()
+    city.setTopTime()
+  }, 1000 * 60 * 60)
```
Comment (saikrishnanp) [inline src/baseClass.js]: Why is this `1000 * 60 * 60`, can't we just use `3600000`.
Also having numbers directly here confuses on what metric is that for. 
Let's store it in a variable and reuse. eg const 1hourInMs = 3600000;

File: src/fetchData.js [hunk 2]
```diff
@@ -4,9 +4,41 @@ import { sortElements } from './sortElements.js'
  * @returns {object} cityDetails
  */
 export async function fetchCityDetails () {
-  let cityDetails = await fetch('../Assets/HTML&CSS/files/data.json')
-    .then((res) => { return res.json() })
-    .then((data) => { return data })
+  const getCityDetailsUrl = 'https://soliton.glitch.me/all-timezone-cities'
```
Comment (saikrishnanp) [inline src/fetchData.js]: This and line 40 can be like this
Also let the base url const be stored globally.
```suggestion
  const weatherApiBaseUrl = 'https://soliton.glitch.me'
  const getCityDetailsUrl = `${weatherApiBaseUrl }/all-timezone-cities`
```

File: src/fetchData.js [hunk 3]
```diff
@@ -4,9 +4,41 @@ import { sortElements } from './sortElements.js'
...
+  let cityDetails = await fetch(getCityDetailsUrl)
```
Comment (saikrishnanp) [inline src/fetchData.js:9]: Did you notice a delay until the city tiles populate in the UI? Can we add a splash screen(loading screen) until then?

File: src/fetchData.js [hunk 4]
```diff
@@ -4,9 +4,41 @@ import { sortElements } from './sortElements.js'
...
+    headers: {
+
+      Accept: 'application/json',
+
+      'Content-Type': 'application/json'
+
+    },
+    body: JSON.stringify(body)
+  }
+  const getNextFiveHourUrl = 'https://soliton.glitch.me/hourly-forecast'
```
Comment (saikrishnanp) [inline src/fetchData.js]: This url can also be formed with const baseUrl

File: src/middleSection.js [hunk 5]
```diff
@@ -89,8 +96,32 @@ const changeMiddleSegment = (cities) => {
...
 }
 
-export const middleSection = { updateScrollButtonsVisibility, addCardsbySpinner, scrollItems, selectIcon, changeMiddleSegment, filterMiddleSegment }
+/**
+Method to scroll Middle section both Horizontally and Vertically
+ * @param {object} target - target element
+ * @param {object} targetContainer - underlying container
+ */
+function scrollEventPassing (target, targetContainer) {
+  target.addEventListener('wheel', (evt) => {
```
Comment (saikrishnanp) [inline src/middleSection.js]: Do we need the eventListener added in function scope? Won't it be needed to be defined in global scope?

File: src/middleSection.js [hunk 6]
```diff
@@ -89,8 +96,32 @@ const changeMiddleSegment = (cities) => {
...
+
+      })
+    }
+
+    if (evt.deltaX !== 0) {
+      targetContainer.scrollLeft += evt.deltaX
+    }
+  })
+}
+scrollEventPassing(document.querySelector('.items-overlay'), document.querySelector('.items-container'))
```
Comment (saikrishnanp) [inline src/middleSection.js:127]: If we're calling the function by default, we can check if it can be used with IIFE

**Review by saikrishnanp — CHANGES_REQUESTED** [review]

File: src/middleSection.js (hunk 5 above)
Comment (Abishek666) [reply src/middleSection.js]: we are giving different value for the function parameter in different calls... i.e card and whole  scroll container as targetcontainer

Comment (saikrishnanp) [inline src/middleSection.js]:
```suggestion
  target.addEventListener('wheel', (event) => {
```

File: src/baseClass.js (hunk 1 above)
Comment (saikrishnanp) [reply src/baseClass.js]: Why were the comments resolved without updating?

**Review by saikrishnanp — CHANGES_REQUESTED** [review]

File: src/middleSection.js (hunk 6 above)
Comment (Abishek666) [reply src/middleSection.js:127]: IIFE functions cant be called outside

**Review by saikrishnanp — APPROVED** [review]
LGTM

**Pull request merged**